mne/tests/test_epochs.py

```diff
@@ -3813,84 +3841,122 @@ def test_epoch_annotations_cases():
+    # 'surround_epoch' should be in the third Epoch only
+    assert 'surround_epoch' in third_epoch_ant
+    assert all('surround_epoch' not in np.array(sublist)
+               for sublist in epoch_ants[:-1])
 
     # 'multiple' should be in 2nd and 3rd Epoch
-    second_epoch_ant = np.array(epoch_ants[1])
-    third_epoch_ant = np.array(epoch_ants[2])
     assert 'multiple' not in first_epoch_ant
     assert 'multiple' in second_epoch_ant
     assert 'multiple' in third_epoch_ant
 
-    # before we don't load the data, bad_noisy Annotation is still part of
-    # Epochs because we haven't dropped bad epochs yet
-    assert 'bad_noisy' in third_epoch_ant
-    epochs.load_data()
-    epoch_ants = epochs.get_annotations_per_epoch()
-    second_epoch_ant = np.array(epoch_ants[1])
-    assert all('bad_noisy' not in np.array(sublist) for sublist in epoch_ants)
-
-    # when preload is passed in, then Epochs overlapping with BAD are
-    # dropped, so 'bad_noisy' will be gone
-    epochs = Epochs(raw, events=events, tmin=-0.5, tmax=0.5, preload=True)
-    epoch_ants = epochs.get_annotations_per_epoch()
-    second_epoch_ant = np.array(epoch_ants[1])
-    assert all('bad_noisy' not in np.array(sublist) for sublist in epoch_ants)
-    epochs_one = epochs.copy()
-
     # if we drop the first Epoch, then some Annotations will now not
     # be part of Epoch Annotations, and others will be shifted
-    epochs = Epochs(raw, events=events, tmin=-0.5, tmax=0.5)
-    epochs.drop(0)
+    epochs = Epochs(raw, events=events, tmin=0, tmax=1, baseline=None)
+    epochs = epochs.drop(0)
     epoch_ants = epochs.get_annotations_per_epoch()
-    assert all('start' not in np.array(sublist) for sublist in epoch_ants)
-    assert all('before' not in np.array(sublist) for sublist in epoch_ants)
-    assert all('stop' not in np.array(sublist) for sublist in epoch_ants)
+    assert all(x not in np.array(sublist) for sublist in epoch_ants for x in [
+        'coincident_offset', 'straddles_onset', 'straddles_offset'])
 
     # 'multiple' should be in 1st and 2nd Epoch now
     first_epoch_ant = np.array(epoch_ants[0])
     second_epoch_ant = np.array(epoch_ants[1])
-    assert 'multiple' in second_epoch_ant
     assert 'multiple' in first_epoch_ant
+    assert 'multiple' in second_epoch_ant
 
-    # test that concatenation does not preserve epochs
+    # test that concatenation does not preserve annotations
+    old_epochs = epochs.copy()
     with pytest.warns(RuntimeWarning, match='Annotations'):
-        concatenate_epochs([epochs, epochs_one])
+        concatenate_epochs([epochs, old_epochs])
+
+    # concatenation should not change the *input* Epochs' annotations
+    assert epochs.annotations == old_epochs.annotations
+
+
+def test_epochs_annotations_backwards_compat():
+    """Test backwards compatibility with Epochs saved without annotations."""
+    # loading an earlier saved file should work
+    fname = op.join(misc.data_path(), 'audio', 'audio-epo.fif')
```

Comment:
This is bad because it will silently download the `misc` dataset for people. We could fix this and have a decorator, but really we don't really want to use the `misc` ataset in testing unless we absolutely must.

Instead of doing this, I would use `monkeypatch` like:
```

def no_sfreq_write_float(a, b, c):
    if b == FIFF.FIFFV_MNE_EPOCHS_RAW_SFREQ:
        return
    return write_tag(a, b, c)

monkeypatch.setattr(mne.epochs, 'write_float', no_sfreq_write_float)
```
Then create and save Epochs. @adam2392 feel free to give it a try, or if it's not clear what I mean I can quickly push a commit to do it

Reply:
@larsoner is this fix what you meant?